Lib/numbers.py

```diff
@@ -203,7 +203,7 @@ def __divmod__(self, other):
         return (self // other, self % other)
 
     def __rdivmod__(self, other):
-        """divmod(other, self): The pair (self // other, self % other).
+        """divmod(other, self): The pair (other // self, other % self).
```

Comment:
The original `__rdivmod__` is somehow correct because note the order of parameters:

- In `__divmod__` it's `self` then `other` for input and output.

- In `__rdivmod__`, they're flipped between input and output.

```python
    def __divmod__(self, other):
        """divmod(self, other): The pair (self // other, self % other).

        Sometimes this can be computed faster than the pair of
        operations.
        """
        return (self // other, self % other)

    def __rdivmod__(self, other):
        """divmod(other, self): The pair (self // other, self % other).

        Sometimes this can be computed faster than the pair of
        operations.
        """
        return (other // self, other % self)
```

It's perhaps a bit confusing the parameters are also flipped between the docstring and code. Maybe have them match?

```suggestion
        """divmod(self, other): The pair (other // self, other % self).
```

Reply:
I am afraid I disagree. `divmod(self, other)` cannot return `(self // other, self % other)` and `(other // self, other % self)` at the same time. Only the former is correct:

```python
divmod(x, y) == (x // y, x % y)
```

Now that the mapping between input and output is clear, here is how it works: `divmod(x, y)` tries `x.__divmod__(y)` and falls back on `y.__rdivmod__(x)` (or the other way round if `type(y)` is a subtype of `type(x)`). Therefore:

```python
divmod(x, y) == x.__divmod__(y)  == (x // y, x % y)
divmod(x, y) == y.__rdivmod__(x) == (x // y, x % y)
```

So if in the *method* calls we rename the first argument to `self` and the second argument to `other`, we get:

```python
divmod(self, other) == self.__divmod__(other)  == (self // other, self % other)
divmod(other, self) == self.__rdivmod__(other) == (other // self, other % self)
```

Reply:
Also consider the doc strings of all the other methods (which are correct) and on which I am basing this pull request. For instance:

```python
    @abstractmethod
    def __floordiv__(self, other):
        """self // other: The floor() of self/other."""
        raise NotImplementedError

    @abstractmethod
    def __rfloordiv__(self, other):
        """other // self: The floor() of other/self."""
        raise NotImplementedError
```

Reply:
Ah right sorry, I misread! It's `divmod` in the docstring.